zio-http-datastar-sdk/src/main/scala/zio/http/datastar/ServerSentEventGenerator.scala

```diff
@@ -0,0 +1,195 @@
+  }
+}
+
+sealed trait ElementPatchMode extends Product with Serializable {
+  override def toString: String = this.productPrefix.toLowerCase
+}
+
+object ElementPatchMode {
+
+  /** Morph entire element, preserving state */
+  case object Outer extends ElementPatchMode
+
+  /** Morph inner HTML only, preserving state */
+  case object Inner extends ElementPatchMode
+
+  /** Replace entire element, reset state */
+  case object Replace extends ElementPatchMode
+
+  /** Insert at beginning inside target */
+  case object Prepend extends ElementPatchMode
+
+  /** Insert at end inside target */
+  case object Append extends ElementPatchMode
+
+  /** Insert before target element */
+  case object Before extends ElementPatchMode
+
+  /** Insert after target element */
+  case object After extends ElementPatchMode
+
+  /** Remove target element from DOM */
+  case object Remove extends ElementPatchMode
+}
+
+final case class PatchElementOptions(
+  selector: Option[CssSelector] = None,
+  mode: ElementPatchMode = ElementPatchMode.Outer,
+  useViewTransition: Boolean = false,
+  eventId: Option[String] = None,
+  retryDuration: Duration = 1000.millis,
+)
+
+object PatchElementOptions {
+  val default: PatchElementOptions = PatchElementOptions()
+}
+
+final case class PatchSignalOptions(
+  onlyIfMissing: Boolean = false,
+  eventId: Option[String] = None,
+  retryDuration: Duration = 1000.millis,
+)
+
+object PatchSignalOptions {
+  val default: PatchSignalOptions = PatchSignalOptions()
+}
+
+final case class ExecuteScriptOptions(
+  autoRemove: Boolean = true,
+  attributes: Seq[(String, String)] = Seq.empty,
+  eventId: Option[String] = None,
+  retryDuration: Duration = 1000.millis,
+)
+
+object ExecuteScriptOptions {
+  val default: ExecuteScriptOptions = ExecuteScriptOptions()
+}
+
+object ServerSentEventGenerator {
+
+  def executeScript(
+    script0: String,
+    options: ExecuteScriptOptions = ExecuteScriptOptions(),
+  ): ZIO[Datastar, Nothing, Unit] = {
+    val removeAttr = if (options.autoRemove) Dom.attr("data-effect", "el.remove") else Dom.empty
+    patchElements(
+      script(Dom.text(script0), removeAttr)(options.attributes.map(a => Dom.attr(a._1, a._2)): _*),
+      PatchElementOptions(eventId = options.eventId, retryDuration = options.retryDuration),
+    )
+  }
+
+  def patchElements(
+    elements: String,
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements.split('\n').map(Dom.raw).toList, options)
+
+  def patchElements(
+    elements: String,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements, PatchElementOptions.default)
+
+  def patchElements(
+    element: Dom,
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] = patchElements(Iterable(element), options)
+
+  def patchElements(
+    element: Dom,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(Iterable(element), PatchElementOptions.default)
```

Comment:
Why using `Iterable` here? 🤔
```suggestion
    patchElements(List(element), PatchElementOptions.default)
```